assets/js/components/dashboard-details/DashboardDetailsEntityHeaderContainer.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * DashboardDetailsEntityHeaderContainer component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Link from '../link';
+import Layout from '../layout/layout';
+
+export default function DashboardDetailsEntityHeaderContainer( { children, permalink } ) {
+	return (
+		<div className="mdc-layout-grid__cell mdc-layout-grid__cell--span-12">
+			<Layout>
+				<div className="mdc-layout-grid">
+					<div className="mdc-layout-grid__inner">
+						<div className="mdc-layout-grid__cell mdc-layout-grid__cell--span-12">
+							{ children }
+							<Link href={ permalink } inherit external>
+								{ permalink }
+							</Link>
```

Comment:
This link shouldn't be there for the `DashboardDetailsEntityNotFoundView`, so it should be removed from here and become part of `children` instead - per the ACs and IB the message should replace both the title and URL, the URL is already part of the message and should not be linked.